apps/minifront/src/components/swap/swap-form/index.tsx

```diff
@@ -23,47 +24,41 @@ export const SwapForm = () => {
   const { onSubmit, submitButtonLabel, duration, submitButtonDisabled } =
     useStoreShallow(swapFormSelector);
 
+  const sharedLayoutId = useId();
+
   return (
-    <Card>
-      <form
-        className='flex flex-col gap-4 xl:gap-3'
-        onSubmit={e => {
-          e.preventDefault();
-          void onSubmit();
-        }}
-      >
-        <TokenSwapInput />
+    <Card layout>
+      <LayoutGroup>
+        <form
+          className='flex flex-col gap-4 xl:gap-3'
+          onSubmit={e => {
+            e.preventDefault();
+            void onSubmit();
+          }}
+        >
+          <TokenSwapInput />
 
-        <InputBlock label='Speed'>
-          <div className='mt-2'>
+          <InputBlock label='Speed' layout>
             <DurationSlider />
-          </div>
-        </InputBlock>
-
-        {duration !== 'instant' && (
-          <InputBlock label='Output'>
-            <div className='mt-2'>
-              <Output />
-            </div>
           </InputBlock>
-        )}
 
-        <div className='mt-3 flex gap-2'>
-          {duration === 'instant' && <SimulateSwapButton />}
+          {duration === 'instant' ? (
+            <SimulateSwap layoutId={sharedLayoutId} />
```

Comment:
The `layoutId` allows these two to have "shared layout transitions," where one appears to morph into the other.

Reply:
Whoa, so that's how it works. Will definitely need a Thursday slot to show folks how this works. Very cool.